File: app/src/main/java/rxProxy/factories/RxFactoriesImpl.kt
package rxProxy.factories

import com.jetbrains.handson.mpp.mobile.rxProxy.factories.Sleeper
import com.jetbrains.handson.mpp.mobile.the_rx.factories.ObservableFactory
import com.jetbrains.handson.mpp.mobile.the_rx.factories.RxFactories

class RxFactoriesImpl: RxFactories {

    private val observableFactory = ObservableFactoryImpl()
    private val schedulerModule = SchedulerModuleImpl()

    override fun getObservableFactory(): ObservableFactory {
        return observableFactory
    }

    override fun getSchedulerModule(): com.jetbrains.handson.mpp.mobile.rxProxy.factories.SchedulerModule {
        return schedulerModule
    }

    override fun getSleeper(): Sleeper {
        return object: Sleeper {
            override fun sleepFor(millisec: Long) {
                Thread.sleep(millisec)
            }
        }
    }
}

File: native/ConceprFirstRxSwift/rxProxy/factories/RxFactoriesImpl.swift
//
//  RxFactoriesImpl.swift
//  ConceprFirstRxSwift
//
//  Created by Marc Gilbert on 06/12/2019.
//  Copyright © 2019 Marc Gilbert. All rights reserved.
//

import Foundation
import SharedCode

class RxFactoriesImpl : RxFactories {
    
    let observableFactory = ObservableFactoryImpl()
    let schedulerModuler = SchedulerModuleImpl()
    let sleeper = SleeperImpl()
    
    func getObservableFactory() -> ObservableFactory {
        return observableFactory
    }
    
    func getSchedulerModule() -> SchedulerModule {
        return schedulerModuler
    }
    
    func getSleeper() -> Sleeper {
        return sleeper
    }
}
